SYNTAX.md

```diff
@@ -30,13 +30,27 @@ These are sections that are specified directly within the `pipeline` argument cl
         * *Type*: bareword
         * *Description*: If given, node/image management will need to be specified explicitly in stages and no 
         automatic `checkout scm` call will occur.
-* *Takes a Closure*: No
+* *Takes a Closure*: yes
 * *Examples*:
-    * `agent label:'has-docker', docker:'ubuntu:lts'`
-    * `agent docker:'ubuntu:lts'`
-    * `agent label:'hi-speed'`
-    * `agent none`
-    * `agent any`
+
+```groovy
+agent {
+    label 'has-docker'
+    docker 'ubuntu:lts'
+}
+
+agent {
+    docker 'ubuntu:lts'
+}
+ 
+agent {
```

Comment:
I see, now I understand what you were referring to before. From my perspective, `none` and `any` could be some agent model that just accepts no and an optional closure, respectively. If I wanted to run on any agent with the label `hi-speed`, I would use `agent any { label: 'hi-speed' }` but if I wanted to run on a docker container, I would use `agent docker { image: 'maven' }` the `docker` agent type would accept a label, also, to filter to agents with specific labels. This would seem to not require any sort of special handling for `none` and `any` and would allow things like the `docker` agent to automatically filter without a label to only docker-enabled agents. Am I missing something? This would be analogous to steps: select the step type, fill out the step parameters (although admittedly that's somewhat of a UI thing)...